test/e2e-kind.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly CT_VERSION=v2.3.3
+readonly KIND_VERSION=v0.5.1
+readonly CLUSTER_NAME=chart-testing
+readonly K8S_VERSION=v1.14.3
+
+run_ct_container() {
+    echo 'Running ct container...'
+    docker run --rm --interactive --detach --network host --name ct \
+        --volume "$(pwd)/test/ct.yaml:/etc/ct/ct.yaml" \
+        --volume "$(pwd):/workdir" \
+        --workdir /workdir \
+        "quay.io/helmpack/chart-testing:$CT_VERSION" \
+        cat
+    echo
+}
+
+cleanup() {
+    echo 'Removing ct container...'
+    docker kill ct > /dev/null 2>&1
+
+    kind delete cluster --name "$CLUSTER_NAME" || true
+
+    echo 'Done!'
+}
+
+docker_exec() {
+    docker exec --interactive ct "$@"
+}
+
+create_kind_cluster() {
+    if ! [ -x "$(command -v kind)" ]; then
+        echo 'Installing kind...'
+
+        curl -sSLo kind "https://github.com/kubernetes-sigs/kind/releases/download/$KIND_VERSION/kind-linux-amd64"
+        chmod +x kind
+        sudo mv kind /usr/local/bin/kind
+    fi
```

Comment:
I wonder if this ends up doing more harm than good. I tried running `make test` on macOS and got stymied for a bit because I didn't realize this installed the Linux binary on my machine :smile:

Also the use of `sudo` inside the installer was a little icky. Maybe better to just error out with a pointer to https://kind.sigs.k8s.io if we can't find `kind`?

Reply:
ah yes - this was left in with an eye on Jenkins but I can just install kind explicitly in the Jenkins job.